crates/re_renderer/src/allocator/cpu_write_gpu_read_belt.rs

```diff
@@ -85,26 +85,47 @@ where
         self.unwritten_element_range.start
     }
 
-    /// Copies the entire buffer to a texture and drops it.
-    pub fn copy_to_texture(
+    /// Copies all so far written data to a rectangle on a single 2d texture layer.
+    ///
+    /// Assumes that the buffer consists of as-tightly-packed-as-possible rows of data.
+    /// (taking into account required padding as specified by [`wgpu::COPY_BYTES_PER_ROW_ALIGNMENT`])
+    ///
+    /// Implementation note:
+    /// Does 2D-only entirely for convenience as it greatly simplifies the input parameters.
+    pub fn copy_to_texture2d(
         self,
         encoder: &mut wgpu::CommandEncoder,
         destination: wgpu::ImageCopyTexture<'_>,
-        bytes_per_row: Option<NonZeroU32>,
-        rows_per_image: Option<NonZeroU32>,
-        copy_size: wgpu::Extent3d,
+        copy_size_width: u32,
+        copy_size_height: u32,
```

Comment:
Not a big fan of the terminology here, `size` makes me think we're dealing with bytes...

I would suggest either `extents: UVec2`, `dimensions: UVec2` or `width: u32, height: u32`

Reply:
took it over from the wgpu parameter name `copy_size`. But makes sense. Not sure about UVec2 vs width+height. It's all over the place in re_renderer atm

Reply:
let's do `extends: glam::UVec2` for all these going forward. Most compact and ticks the boxes